src/vehicle.cpp

```diff
@@ -5778,6 +5778,20 @@ void vehicle::enable_refresh()
     refresh();
 }
 
+void vehicle::refresh_active_item_cache()
+{
+    // Need to manually backfill the active item cache since the part loader can't call its vehicle.
+    for( const vpart_reference &vp : get_any_parts( VPFLAG_CARGO ) ) {
```

Comment:
It looks like this can added as a part of the loop in `vehicle::refresh` instead of a separate loop, something like this:

```diff
diff --git a/src/vehicle.cpp b/src/vehicle.cpp
index ac704ba924..14be259663 100644
--- a/src/vehicle.cpp
+++ b/src/vehicle.cpp
@@ -5778,20 +5778,6 @@ void vehicle::enable_refresh()
     refresh();
 }
 
-void vehicle::refresh_active_item_cache()
-{
-    // Need to manually backfill the active item cache since the part loader can't call its vehicle.
-    for( const vpart_reference &vp : get_any_parts( VPFLAG_CARGO ) ) {
-        auto it = vp.part().items.begin();
-        auto end = vp.part().items.end();
-        for( ; it != end; ++it ) {
-            if( it->needs_processing() ) {
-                active_items.add( *it, vp.mount() );
-            }
-        }
-    }
-}
-
 /**
  * Refreshes all caches and refinds all parts. Used after the vehicle has had a part added or removed.
  * Makes indices of different part types so they're easy to find. Also calculates power drain.
@@ -5980,6 +5966,14 @@ void vehicle::refresh( const bool remove_fakes )
         if( vpi.has_flag( VPFLAG_ENABLED_DRAINS_EPOWER ) ) {
             accessories.push_back( p );
         }
+        if( vpi.has_flag( VPFLAG_CARGO ) ) {
+            // Need to manually backfill the active item cache since the part loader can't call its vehicle.
+            for( auto it = vp.part().items.begin(); it != vp.part().items.end(); ++it ) {
+                if( it->needs_processing() ) {
+                    active_items.add( *it, vp.mount() );
+                }
+            }
+        }
     }
 
     rail_wheel_bounding_box.p1 = point( railwheel_xmin, railwheel_ymin );
@@ -6078,7 +6072,6 @@ void vehicle::refresh( const bool remove_fakes )
     zones_dirty = true;
     invalidate_mass();
     occupied_cache_pos = { -1, -1, -1 };
-    refresh_active_item_cache();
 }
 
 vpart_edge_info vehicle::get_edge_info( const point &mount ) const
diff --git a/src/vehicle.h b/src/vehicle.h
index 60a169a8b8..a05ed1f3ef 100644
--- a/src/vehicle.h
+++ b/src/vehicle.h
@@ -837,9 +837,6 @@ class vehicle
         //Refresh all caches and re-locate all parts
         void refresh( bool remove_fakes = true );
 
-        // Refresh active_item cache for vehicle parts
-        void refresh_active_item_cache();
-
         /**
          * Set stat for part constrained by range [0,durability]
          * @note does not invoke base @ref item::on_damage callback
```

Reply:
@dseguin I can try to make that change asap and see if it still works. I think it might not though since the part that was causing `active_items` to become empty was further down in the method in [this part](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/src/vehicle.cpp#L6032-L6035):

```
// add all the obstacles first
for( const std::pair <const point, std::vector<int>> &rp : relative_parts ) {
    add_fake_part( rp.first, "OBSTACLE" );
}
```

If you check `active_items.get_for_processing()` before that loop, it will have the expected items. Afterwards it's empty. Still not sure why though.

Reply:
That's strange, I don't see anything that would clear the cache there :thinking:
If it causes issues then feel free to ignore the suggestion.